auth/src/main/java/com/firebase/ui/auth/ui/HelperActivityBase.java

```diff
@@ -42,6 +51,40 @@ protected void onCreate(Bundle savedInstance) {
         super.onCreate(savedInstance);
         mAuthHelper = new AuthHelper(getFlowParams());
         mProgressDialogHolder = new ProgressDialogHolder(this);
+
+        getFlowHolder().getIntentStarter().observe(this, new Observer<Pair<Intent, Integer>>() {
```

Comment:
Since we're already using data types like `ActivityResult` let's create a custom object type like `IntentRequest` rather than use a `Pair` here.

Reply:
Yep, that's just me being lazy. 😄